packages/esm-patient-banner-app/src/banner/patient-banner.component.tsx

```diff
@@ -9,25 +9,43 @@ import CustomOverflowMenuComponent from '../ui-components/overflow-menu.componen
 import styles from './patient-banner.scss';
 import { useTranslation } from 'react-i18next';
 import { Button } from 'carbon-components-react';
-import { ExtensionSlot, age } from '@openmrs/esm-framework';
+import { ExtensionSlot, age, navigate } from '@openmrs/esm-framework';
 
 interface PatientBannerProps {
   patient: fhir.Patient;
   patientUuid: string;
+  patientUrl?: string;
+  hidePatientSearchPanel?: () => void;
 }
 
-const PatientBanner: React.FC<PatientBannerProps> = ({ patient, patientUuid }) => {
+const PatientBanner: React.FC<PatientBannerProps> = ({ patient, patientUuid, patientUrl, hidePatientSearchPanel }) => {
```

Comment:
The patient banner should not know anything about the context(s) in which it is used. I suggest the following API:

```suggestion
const PatientBanner: React.FC<PatientBannerProps> = ({ patient, patientUuid, onClick }) => {
```

and then both the `navigate` call and the `hidePatientSearchPanel` call would be part of the `onClick` handler passed in by patient search.

Reply:
Updated